GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -700,27 +697,28 @@ private void SetShortcutKeyDisplayStringsFromHotkeySettings()
 
         private void RefreshWorkingDirCombo()
         {
-            Repository r = null;
-            if (Repositories.RepositoryHistory.Repositories.Count > 0)
+            if (!Module.WorkingDir.Equals(_repositoryHistory.FirstOrDefault()?.Path, StringComparison.InvariantCultureIgnoreCase))
             {
-                r = Repositories.RepositoryHistory.Repositories[0];
-            }
+                var path = Module.WorkingDir;
 
-            List<RecentRepoInfo> mostRecentRepos = new List<RecentRepoInfo>();
-
-            if (r == null || !r.Path.Equals(Module.WorkingDir, StringComparison.InvariantCultureIgnoreCase))
-            {
-                Repositories.AddMostRecentRepository(Module.WorkingDir);
+                // save the current repository as the most recent
+                // the call is blocking because we are reloading the list of repositories
+                ThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
❗️ Use the following form instead since it needs to be synchronous:

```csharp
_repositoryHistory = ThreadHelper.JoinableTaskFactory.Run(
    () => RepositoryHistoryManager.Locals.AddAsMostRecentAsync(path));
```